src/uint/encoding.rs

```diff
@@ -172,6 +190,22 @@ const fn decode_hex_byte(bytes: [u8; 2]) -> u8 {
     result
 }
 
+/// Check that the input consists of hexadecimal characters,
+/// short-circuiting on invalid input.
+const fn validate_hex(bytes: &[u8]) -> Option<usize> {
+    if bytes.len() % 2 == 1 {
+        return None;
+    }
+    let mut i = 0;
+    while i < bytes.len() {
+        if !matches!(bytes[i], b'0'..=b'9' | b'a' ..= b'f' | b'A'..=b'F') {
+            return None;
+        }
+        i += 1;
+    }
+    Some(bytes.len() / 2)
+}
```

Comment:
This is not constant-time.

Perhaps you could change it to be? Or otherwise you need to make the methods `*_vartime` to honor the "constant time by default" contract of this library, but I would prefer you just make the implementation constant-time instead.

It should also return `CtOption` rather than `Option`.

Reply:
I don't really see how this can be constant time overall, since the implementation short circuits on invalid input, and I'm not sure what the alternative would be. It would also be nice for it to handle leading zeros on hex values, but in that case is it meant to be constant time in the input length or in the number's representation?

Reply:
> the implementation short circuits on invalid input, and I'm not sure what the alternative would be.

You can defer checking an error condition until the end of the data, which is what `from_be_hex` does already:

https://github.com/RustCrypto/crypto-bigint/blob/2122b4a/src/uint/encoding.rs#L55-L56